pkg/cliplugins/kubestellar/get_log/cmd/cmd.go

```diff
@@ -0,0 +1,209 @@
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	kubeconfig   string
+	followFlag   bool
+	linesFlag    string
+	whichArg     string
+	kubectlFlags []string
+	ksNamespace  string
+)
+
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get_log [flags] which",
+		Short: "Get the log output from a central KubeStellar or kcp process",
+		Args:  cobra.ExactArgs(1),
+		Run:   getLog,
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&kubeconfig, "kubeconfig", "c", "", "Path to the kubeconfig file")
+	flags.BoolVarP(&followFlag, "follow", "f", false, "Follow the log output")
+	flags.StringVarP(&linesFlag, "lines", "n", "+0", "Number of lines to display")
+	flags.StringArrayVar(&kubectlFlags, "kubectl-flag", []string{}, "Additional kubectl flags")
+
+	return cmd
+}
+
+func getLog(cmd *cobra.Command, args []string) {
+	whichArg = args[0]
+	ksNamespace = "kubestellar"
+
+	logfile := ""
+	switch whichArg {
+	case "kcp":
+		logfile = "kcp.log"
+	case "where-resolver":
+		logfile = "kubestellar-where-resolver-log.txt"
+	case "mailbox-controller":
+		logfile = "mailbox-controller-log.txt"
+	case "placement-translator":
+		logfile = "placement-translator-log.txt"
+	default:
+		fmt.Println("Argument 1 must be one of: kcp, where-resolver, mailbox-controller, placement-translator")
+		os.Exit(1)
+	}
+
+	// Build the Kubernetes clientset
+	config, err := buildConfig()
+	if err != nil {
+		fmt.Println("Error building Kubernetes config:", err)
+		os.Exit(1)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Println("Error creating Kubernetes clientset:", err)
+		os.Exit(1)
+	}
+	namespace, err := getNamespaceNames(clientset, ksNamespace)
+	if err != nil {
+		fmt.Println("Error getting server namespace:", err)
+		os.Exit(1)
+	}
+	podName, err := getServerPodName(clientset, namespace)
+	if err != nil {
+		fmt.Println("Error getting server pod name:", err)
+		os.Exit(1)
+	}
+
+	for {
+		if podName != "" {
+			podName = strings.Trim(string(podName), "'")
+			break
+		}
+		time.Sleep(10 * time.Second)
+	}
+
+	err = executeCommandInPod(clientset, namespace, config, podName, []string{"ls", "/home/kubestellar/ready"}, os.Stdin, os.Stdout, os.Stderr)
+	if err != nil {
+		fmt.Println("Error executing command in pod:", err)
+		os.Exit(1)
+	}
+
+	kubectlLogArgs := []string{"tail", "kubestellar-logs/" + logfile}
+	if followFlag {
+		kubectlLogArgs = append(kubectlLogArgs, "-f")
+	}
+	kubectlLogArgs = append(kubectlLogArgs, "-n", linesFlag)
+
+	err = executeCommandInPod(clientset, namespace, config, podName, kubectlLogArgs, os.Stdin, os.Stdout, os.Stderr)
+	if err != nil {
+		fmt.Println("Error executing command in pod:", err)
+		os.Exit(1)
+	}
+
+}
+
+func buildConfig() (*rest.Config, error) {
+	if kubeconfig == "" {
+		if home := homedir.HomeDir(); home != "" {
+			kubeconfig = home + "/.kube/config"
+		}
+	}
+	return clientcmd.BuildConfigFromFlags("", kubeconfig)
+}
```

Comment:
I think we want to be invoking standard library code to do stuff like this, rather than replicate this small subset of that functionality here.

I was hoping that we could ride on the main functionality of `k8s.io/cli-runtime/pkg/genericclioptions`, not just its streams object.  See https://github.com/kubestellar/kubestellar/blob/v0.6.0/cmd/kube-watchall/main.go#L55-L56 and https://github.com/kubestellar/kubestellar/blob/v0.6.0/cmd/kube-watchall/main.go#L79 for the powerful stuff.  This is where the standard set of kubectl flags is defined and used.